### PR TITLE
Upgrade to react-query v4

### PR DESCRIPTION
**Pull request by andersrognstad**
Se https://tanstack.com/query/v4/docs/react/guides/migrating-to-react-query-4 / https://fwuensche.medium.com/react-query-migrating-to-v4-5ae34f74caf3

### CODE REVIEW

**Comment by geir-waagboe** [comment]
Files changed 115!

File: src/components/AktivBrukerTilgangLaster.tsx [hunk 1]
```diff
@@ -17,7 +17,7 @@ const AktivBrukerTilgangLaster = ({
   children,
 }: AktivBrukerTilgangLasterProps): ReactElement => {
   const {
-    isLoading: henterTilgang,
+    isInitialLoading: henterTilgang,
```
Comment (andersrognstad) [inline src/components/AktivBrukerTilgangLaster.tsx:20]: Endret til isInitialLoading for queries som kan være disabled (Ref: https://tanstack.com/query/v4/docs/react/guides/migrating-to-react-query-4#disabled-queries https://tanstack.com/query/v4/docs/react/guides/disabling-queries#isinitialloading)

File: src/data/motebehov/useBehandleMotebehov.ts [hunk 2]
```diff
@@ -32,7 +32,7 @@ export const useBehandleMotebehov = () => {
     },
     onSettled: () =>
       queryClient.invalidateQueries(motebehovQueryKey, {
-        refetchActive: false,
+        refetchType: "none",
```
Comment (andersrognstad) [inline src/data/motebehov/useBehandleMotebehov.ts:35]: https://tanstack.com/query/v4/docs/react/guides/migrating-to-react-query-4#refetchactive--refetchinactive

File: src/index.js [hunk 3]
```diff
@@ -17,7 +17,11 @@ Sentry.init({
 
 const queryClient = new QueryClient({
   defaultOptions: {
+    mutations: {
```
Comment (andersrognstad) [inline src/index.js:20]: Tilsvarer v3 (https://tanstack.com/query/v4/docs/react/guides/migrating-to-react-query-4#queries-and-mutations-per-default-need-network-connection-to-run)

File: test/setup.js [hunk 4]
```diff
@@ -13,14 +12,6 @@ const { JSDOM } = require("jsdom");
 const jsdom = new JSDOM("<!doctype html><html><body></body></html>");
 const { window } = jsdom;
 
-setLogger({
```
Comment (andersrognstad) [inline test/setup.js:16]: Flyttet dette til testQueryClient (https://tanstack.com/query/v4/docs/react/guides/migrating-to-react-query-4#setlogger-is-removed)

**Review by geir-waagboe — APPROVED** [review]

**Review by JMLindseth — APPROVED** [review]

**Review by eirikdahlen — APPROVED** [review]

**Pull request merged**